src/modules/m_sslinfo.cpp

```diff
@@ -373,8 +373,28 @@ class ModuleSSLInfo
 		{
 			OperInfo* ifo = i->second;
 			std::string fp = ifo->oper_block->getString("fingerprint");
-			if (MatchFP(cert, fp) && ifo->oper_block->getBool("autologin"))
-				user->Oper(ifo);
+			if (MatchFP(cert, fp))
+			{
+				bool do_login = false;
+				const std::string autologin = ifo->oper_block->getString("autologin");
+				if (stdalgo::string::equalsci(autologin, "if-host-match"))
+				{
+					const std::string userHost = localuser->ident + "@" + localuser->GetRealHost();
+					const std::string userIP = localuser->ident + "@" + localuser->GetIPString();
+					if (InspIRCd::MatchMask(ifo->oper_block->getString("host"), userHost, userIP)) {
+						do_login = true;
+					}
+				}
+				else if (ifo->oper_block->getBool("autologin"))
+				{
+					do_login = true;
+				}
+
+				if (do_login)
+				{
+					user->Oper(ifo);
+				}
```

Comment:
This if block can be without braces as it's a single line and not part of another block (like the else if above is).

Reply:
Done.